sycl/include/sycl/ext/oneapi/device_global/device_global.hpp

```diff
@@ -63,14 +68,49 @@ class device_global_base {
   pointer_t get_ptr() noexcept { return usmptr; }
   pointer_t get_ptr() const noexcept { return usmptr; }
 
+  template <typename, typename, typename> friend class device_global_base;
+
+#ifndef __SYCL_DEVICE_ONLY__
+  template <typename OtherT, typename OtherProps>
+  static constexpr const T &
+  ExtractInitialVal(const device_global_base<OtherT, OtherProps> &Other) {
+    if constexpr (OtherProps::template has_property<device_image_scope_key>())
+      return Other.val;
+    else
+      return Other.init_val;
```

Comment:
I think that's a valid concern. I don't remember if there was a reason not to store it elsewhere. I will investigate if the device global entry list could be used to store the init values instead. I think it might be possible, but I think it would be better as a follow-up patch.

Reply:
Yes, addressing it in a follow-up PR is OK.